snapcraft/plugins/v1/crystal.py

```diff
@@ -92,6 +93,12 @@ def __init__(self, name, options, project):
             ]
         )
 
+    def pull(self):
+        super().pull()
+
+        with contextlib.suppress(FileNotFoundError):
+            shutil.rmtree(os.path.join(self.sourcedir, "bin"))
```

Comment:
Should we remove "lib" as well? To make sure the build isn't stale for all shards.

Reply:
Yea probably.  Waiting to hear back from @sergiusens regarding https://bugs.launchpad.net/snapcraft/+bug/1936671/comments/8.  I.e. if this PR is obsolete.

Reply:
Sorry for the delay, I have replied in the bug